src/main/java/de/tum/in/www1/artemis/web/rest/lecture/AttachmentUnitResource.java

```diff
@@ -144,4 +150,51 @@ public ResponseEntity<AttachmentUnit> createAttachmentUnit(@PathVariable Long le
 
         return ResponseEntity.created(new URI("/api/attachment-units/" + savedAttachmentUnit.getId())).body(savedAttachmentUnit);
     }
+
+    /**
+     * POST lectures/:lectureId/attachment-units/split : creates new attachment units.
+     *
+     * @param lectureId            the id of the lecture to which the attachment units should be added
+     * @param lectureUnitInformationDTO the units that should be created
+     * @param file                 the file to be splitted
+     * @return the ResponseEntity with status 200 (ok) and with body the newly created attachment units
+     */
+    @PostMapping("lectures/{lectureId}/attachment-units/split")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<List<AttachmentUnit>> createAttachmentUnits(@PathVariable Long lectureId, @RequestPart LectureUnitInformationDTO lectureUnitInformationDTO,
+            @RequestPart MultipartFile file) {
+        log.debug("REST request to create AttachmentUnits {} with lectureId {}", lectureUnitInformationDTO, lectureId);
+
+        Lecture lecture = lectureRepository.findByIdWithLectureUnitsElseThrow(lectureId);
+        if (lecture.getCourse() == null) {
+            throw new ConflictException("Specified lecture is not part of a course", "AttachmentUnit", "courseMissing");
+        }
```

Comment:
Can this even happen? Shouldn’t this only happen in case the database query has been implemented wrong and does not fetch the course?

Reply:
I was not sure about this, we have this check when creating a single lecture unit so i also do this check here.